### PR TITLE
Change odometry_data_ to MapByTime.

### PR DESCRIPTION
**Pull request by wohe**

### CODE REVIEW

File: cartographer/mapping_2d/sparse_pose_graph/optimization_problem.cc [hunk 1]
```diff
@@ -246,6 +228,45 @@ const sensor::MapByTime<sensor::ImuData>& OptimizationProblem::imu_data()
...
+      Interpolate(transform::TimestampedTransform{prev_it->time, prev_it->pose},
+                  transform::TimestampedTransform{it->time, it->pose}, time)
+          .transform);
+}
+
+transform::Rigid3d OptimizationProblem::ComputeRelativePose(
+    const int trajectory_id, const NodeData& first_node_data,
+    const NodeData& second_node_data) const {
+  if (odometry_data_.HasTrajectory(trajectory_id)) {
+    std::unique_ptr<transform::Rigid3d> first_node_odometry =
```
Comment (ojura) [inline cartographer/mapping_2d/sparse_pose_graph/optimization_problem.cc]: `const` for unique pointers, also in 3D?

File: cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc [hunk 2]
```diff
@@ -447,6 +436,41 @@ const sensor::MapByTime<sensor::ImuData>& OptimizationProblem::imu_data()
...
+  if (odometry_data_.HasTrajectory(trajectory_id)) {
+    const std::unique_ptr<transform::Rigid3d> first_node_odometry =
+        InterpolateOdometry(trajectory_id, first_node_data.time);
+    const std::unique_ptr<transform::Rigid3d> second_node_odometry =
+        InterpolateOdometry(trajectory_id, second_node_data.time);
+    if (first_node_odometry != nullptr && second_node_odometry != nullptr) {
+      return first_node_odometry->inverse() * (*second_node_odometry);
+    }
+  }
+  // No odometry is available here, so we use the local SLAM result.
```
Comment (cschuet) [inline cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc]: Please document this behavior as a comment before the function declaration.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**